libcoveofds/geojson.py

```diff
@@ -223,32 +224,86 @@ def _process_network(self, geojson_feature_node_or_span: dict) -> None:
             and "network" in geojson_feature_node_or_span["properties"]
         ):
             network = geojson_feature_node_or_span["properties"]["network"]
-            if network.get("id"):
-                # TODO check for inconsistent data here!
-                self._networks[network.get("id")] = copy.deepcopy(network)
-                self._networks[network.get("id")]["nodes"] = []
-                self._networks[network.get("id")]["spans"] = []
-                self._networks[network.get("id")]["phases"] = {}
-                self._networks[network.get("id")]["organisations"] = {}
-
-                # Sort references to phases in contracts
-                if "contracts" in self._networks[network.get("id")] and isinstance(
-                    self._networks[network.get("id")]["contracts"], list
-                ):
-                    for contract in self._networks[network.get("id")]["contracts"]:
-                        if "relatedPhases" in contract and isinstance(
-                            contract["relatedPhases"], list
+            if isinstance(network, dict):
+                network_id = network.get("id")
+                if isinstance(network_id, str) and network_id:
+
+                    # TODO nodes/spans/phases/organisations should not be set in network - check this and warn if so
+
+                    # Is data already seen?
+                    if network_id in self._networks:
+                        # Is it inconsistent with what we have seen before?
+                        if json_diff_function(
+                            self._networks[network_id]["network_data_original"], network
+                        ):
+                            # record error for later
+                            self._inconsistent_network_ids_seen.add(network_id)
+
+                            # Check references to phases in contracts are consistent with what we have seen before.
+                            # This will give a more specific error on the phase_id.
+                            # (However, 2 errors will be recorded - one for network_id and one for phrase_id!
```

Comment:
I think this should be `phase_id`?

Reply:
Ah phrase_id -> phase_id

Reply:
added commit to fix, rebased so behind release commit